pkg/csi/service/common/vsphereutil.go

```diff
@@ -322,6 +358,45 @@ func CreateFileVolumeUtil(ctx context.Context, clusterFlavor cnstypes.CnsCluster
 		}
 	}
 
+	if spec.StoragePolicyID != "" {
+
+		// Check storage policy compatibility.
```

Comment:
Can we de-dupe code? We have almost same code for block and file volume.

Reply:
+1 on this.

Reply:
+1. Also try to check if you can call this function in `topology.go` - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/commonco/k8sorchestrator/topology.go#L958-L994

Reply:
Tested with topology setup and for WCP as well